rocks-db/src/bin/column_copier/main.rs

```diff
@@ -75,38 +76,60 @@ async fn copy_column_families(
         MigrationState::Last,
     )
     .map_err(|e| e.to_string())?;
+    let mut set = JoinSet::new();
 
-    let tasks = columns_to_copy.iter().cloned().map(|cf_name| {
+    // Create a MultiProgress to manage multiple progress bars
+    let m = Arc::new(MultiProgress::new());
+
+    columns_to_copy.iter().cloned().for_each(|cf_name| {
         let sdb = source_db.db.clone();
         let ddb = destination_db.db.clone();
-        tokio::task::spawn_blocking(move || {
+        let m_clone = m.clone();
+
+        set.spawn(tokio::task::spawn_blocking(move || {
+            // Create a progress bar for this column
+            let pb = m_clone.add(ProgressBar::new_spinner());
+            pb.set_style(
+                ProgressStyle::default_spinner()
+                    .template("{spinner:.green} {msg}")
+                    .unwrap(),
+            );
+            pb.set_message(format!("Copying column {}", cf_name));
+            pb.enable_steady_tick(Duration::from_millis(100));
             let start_column = Instant::now();
             let mut iter = sdb.raw_iterator_cf(&sdb.cf_handle(&cf_name).unwrap());
             iter.seek_to_first();
             let mut batch = rocksdb::WriteBatchWithTransaction::<false>::default();
             let dest_handle = &ddb.cf_handle(&cf_name).unwrap();
+            let mut batch_count = 0;
             while iter.valid() {
                 batch.put_cf(dest_handle, iter.key().unwrap(), iter.value().unwrap());
                 if batch.len() >= BATCH_SIZE {
                     if let Err(e) = ddb.write(batch) {
                         tracing::error!("Error copying {}: {}", cf_name, e);
                     }
                     batch = rocksdb::WriteBatchWithTransaction::<false>::default();
+                    batch_count += 1;
+                    // Update the progress bar
+                    pb.set_message(format!(
+                        "Copying column {} - batches copied: {}",
+                        cf_name, batch_count
+                    ));
                 }
                 iter.next();
             }
             if let Err(e) = ddb.write(batch) {
                 tracing::error!("Error copying {}: {}", cf_name, e);
             }
-            println!(
-                "Migrated {} column in {} seconds",
+            pb.finish_with_message(format!(
+                "Finished copying column {} in {} seconds",
                 cf_name,
                 start_column.elapsed().as_secs()
-            );
-        })
+            ));
+        }));
     });
-    for task in tasks {
-        task.await.unwrap();
+    for _ in 0..columns_to_copy.len() {
```

Comment:
You already awaiting the function in the main(), it cannot be executed partially. Therefore, it's redundant.

Reply:
but this function will exit immediately, right? we should join the set somehow, shouldn't we?